plugins/circleci/src/circleci-config.ts

```diff
@@ -102,15 +110,31 @@ export default abstract class CircleCiConfigHook extends Hook<CircleCIState> {
   }
 
   async commitInstall(state: CircleCIState): Promise<void> {
-    const config = await this.getCircleConfig()
-    if (config && !isAutomatedConfig(config)) {
+    const rawConfig = await this.getCircleConfig()
+    if (rawConfig && !isAutomatedConfig(rawConfig)) {
+      const config = YAML.parse(rawConfig)
+      const { jobs, nightlyJobs } = getWorkflowJobs(config)
+      const flattenJob = (job: Job): string => (typeof job === 'string' ? job : Object.keys(job)[0])
+      const flatJobs = jobs?.map(flattenJob) ?? []
+      const flatNightlyJobs = nightlyJobs?.map(flattenJob) ?? []
+      const missingJobs = Object.keys(state.jobs).filter((job) => !flatJobs.includes(job))
+      const missingNightlyJobs = Object.keys(state.nightlyJobs).filter(
+        (job) => !flatNightlyJobs.includes(job)
+      )
+
       const formatMissingHooks = (hooks: string[]): string => hooks.map(styles.hook).join(', ')
       throw new ToolKitError(
-        `Please update your CircleCI config to include the ${formatMissingHooks(
-          Object.keys(state.jobs)
-        )} jobs in the ${styles.heading('tool-kit')} workflow and the ${formatMissingHooks(
-          Object.keys(state.nightlyJobs)
-        )} jobs in the ${styles.heading('nightly')} workflow`
+        `Please update your CircleCI config to include ${
+          missingJobs.length > 0
+            ? `the ${formatMissingHooks(missingJobs)} job(s) in the ${styles.heading('tool-kit')} workflow`
+            : ''
+        }${missingJobs.length > 0 && missingNightlyJobs.length > 0 ? ' and ' : ''}${
+          missingNightlyJobs.length > 0
+            ? `the ${formatMissingHooks(missingNightlyJobs)} job(s) in the ${styles.heading(
+                'nightly'
+              )} workflow`
+            : ''
+        }`
       )
```

Comment:
**thought (non-blocking):** Can we move some of this logic into its' own method to reduce the size of this conditional block?